### PR TITLE
Add memo jtx funclet:

### PR DESCRIPTION
**Pull request by vinniefalco**
* Add Env::trace
* Log parse errors during signing

### CODE REVIEW

File: src/ripple/test/jtx/impl/Env_test.cpp [hunk 1]
```diff
@@ -414,6 +414,22 @@ class Env_test : public beast::unit_test::suite
...
+        Env env(*this);
+        env.fund(XRP(10000), "alice");
+        env(noop("alice"), memodata("data"));
+        env(noop("alice"), memoformat("format"));
+        env(noop("alice"), memotype("type"));
+        env(noop("alice"), memondata("format", "type"));
+        env(noop("alice"), memonformat("data", "type"));
+        env(noop("alice"), memontype("data", "format"));
+        env(noop("alice"), memo("data", "format", "type"));
+        env(noop("alice"), memo("data1", "format1", "type1"), memo("data2", "format2", "type2"));
```
Comment (ximinez) [inline src/ripple/test/jtx/impl/Env_test.cpp]: This test would be better if it somehow verified that the memo was in the ledger for the transactions in addition to implicitly verifying that the transaction succeeded.

Comment (vinniefalco) [inline src/ripple/test/jtx/impl/Env_test.cpp]: The purpose of `Env_test` is to test the jtx framework, not to test memos. This just makes sure that all the funclets compile correctly. A full blown memo test is certainly something we could use - thanks for volunteering to write it!

**Comment by scottschurr** [comment]
This all looks pretty reasonable to me.  The only real downside I see is that the `memo` funclet takes three identical arguments (all `std::string`) and the caller has to know that the first argument is the data, the second the format, and the third the type.  But this interface is just for unit tests.  The caller can solve the ambiguity with a bit of research.

:+1:

**Comment by vinniefalco** [comment]
@scottschurr I hear ya about the 3 arguments, lesser of the evils

**Comment by miguelportilla** [comment]
:+1:

File: src/ripple/test/jtx/impl/memo.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+{
+    auto& jv = jt.jv;
+    auto& ma = jv["Memos"];
+    auto& mi = ma[ma.size()];
+    auto& m = mi["Memo"];
+    m["MemoType"] = strHex(s_);
+}
+
+void
+memondata::operator()(Env const&, JTx& jt) const
```
Comment (sublimator) [inline src/ripple/test/jtx/impl/memo.cpp]: memondata means memo with no data?

Comment (vinniefalco) [inline src/ripple/test/jtx/impl/memo.cpp]: Yerp

Comment (sublimator) [inline src/ripple/test/jtx/impl/memo.cpp]: lol Johnny Memondata

Comment (vinniefalco) [inline src/ripple/test/jtx/impl/memo.cpp]: Great book and movie

Comment (sublimator) [inline src/ripple/test/jtx/impl/memo.cpp]: Shoulda went with memontype ... another death bed regret for syllabic fail

**Pull request closed without merging**